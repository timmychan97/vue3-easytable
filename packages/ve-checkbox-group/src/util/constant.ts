export const PREFIX_CLS = 've-checkbox-group-'

export const EMIT_EVENTS = {
  ON_CHECKED_CHANGE: 'on-checked-change',
}

export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
