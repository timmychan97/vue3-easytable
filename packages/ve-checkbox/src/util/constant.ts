export const PREFIX_CLS = 've-checkbox-'

export const EMIT_EVENTS = {
  ON_CHECKED_CHANGE: 'on-checked-change',
}

export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
