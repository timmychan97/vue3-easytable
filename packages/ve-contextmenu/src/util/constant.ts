export const PREFIX_CLS = 've-contextmenu-'

export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

export const CONTEXTMENU_NODE_TYPES = {
  SEPARATOR: 'SEPARATOR',
}

export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
