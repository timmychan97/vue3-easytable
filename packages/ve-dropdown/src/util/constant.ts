export const PREFIX_CLS = 've-dropdown-'

export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
