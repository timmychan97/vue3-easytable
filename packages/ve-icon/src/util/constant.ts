export const PREFIX_CLS = 've-icon-'

export const EMIT_EVENTS = {}

export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
