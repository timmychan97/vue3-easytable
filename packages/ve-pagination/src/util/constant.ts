export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
