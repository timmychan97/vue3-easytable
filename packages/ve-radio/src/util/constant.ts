export const PREFIX_CLS = 've-radio-'

export const EMIT_EVENTS = {
  ON_RADIO_CHANGE: 'on-radio-change',
}

export const COMPS_NAME = {
  VE_RADIO: 'VeRadio',
}
