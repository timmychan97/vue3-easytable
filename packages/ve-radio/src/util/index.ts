import { PREFIX_CLS } from './constant'

export function clsName(cls: string) {
  return PREFIX_CLS + cls
}
