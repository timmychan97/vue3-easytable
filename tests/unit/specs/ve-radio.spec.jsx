import { later, wrapMount as mount } from '../util'
import veRadio from '@/ve-radio'

describe('veRadio', () => {
  it('render', () => {
    const wrapper = mount({
      render() {
        const { checked, unChecked } = this
        return (
          <div>
            <div>
              <veRadio modelValue={checked}>normal</veRadio>
            </div>
            <div>
              <veRadio disabled modelValue={checked}>
                disabled checked
              </veRadio>
            </div>
            <div>
              <veRadio disabled modelValue={unChecked}>
                disabled unChecked
              </veRadio>
            </div>
            <div>
              <veRadio isControlled isSelected={checked}>
                controlled
              </veRadio>
            </div>
          </div>
        )
      },
      data() {
        return {
          checked: true,
          unChecked: false,
        }
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  it('modelValue prop', async () => {
    const wrapper = mount(veRadio, {
      propsData: {
        modelValue: false,
      },
    })

    expect(wrapper.find('.ve-radio-checked').exists()).toBe(false)

    wrapper.setProps({ isSelected: true })

    await later()
    expect(wrapper.find('.ve-radio-checked').exists()).toBe(false)
    wrapper.unmount()
  })

  it('disable prop', () => {
    const wrapper = mount(veRadio, {
      propsData: {
        modelValue: true,
      },
    })

    expect(wrapper.find('.ve-radio-checked').exists()).toBe(true)
    wrapper.unmount()
  })

  it('label prop', () => {
    const wrapper = mount(veRadio, {
      propsData: {
        modelValue: false,
        label: 'test',
      },
    })

    expect(wrapper.find('.ve-radio-label').text()).toContain('test')
    wrapper.unmount()
  })

  it('isControlled prop', async () => {
    const wrapper = mount(veRadio, {
      propsData: {
        isControlled: true,
        isSelected: false,
      },
    })

    expect(wrapper.find('.ve-radio-checked').exists()).toBe(false)

    wrapper.setProps({ isSelected: true })

    await later()
    expect(wrapper.find('.ve-radio-checked').exists()).toBe(true)
    wrapper.unmount()
  })

  /*   it("click event", async () => {
        const wrapper = mount(veRadio, {
            propsData: {
                value: false,
                label: "test"
            }
        });

        wrapper.trigger("click");

        await later();
        expect(wrapper.find(".ve-radio-checked").exists()).toBe(true);
    }); */

  it('on-radio-change event', async () => {
    const wrapper = mount(veRadio, {
      propsData: {
        modelValue: false,
        label: 'test',
      },
    })

    wrapper.trigger('click')

    await later()
    expect(wrapper.emitted('on-radio-change').length).toEqual(1)
    expect(wrapper.emitted('on-radio-change')[0]).toEqual([true])
    wrapper.unmount()
  })
})
