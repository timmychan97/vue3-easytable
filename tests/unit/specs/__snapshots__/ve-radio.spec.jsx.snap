// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`veRadio > render 1`] = `
<div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-checked"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">normal</span></label></div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-checked ve-radio-disabled"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">disabled checked</span></label></div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-disabled"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">disabled unChecked</span></label></div>
  <div><label class="ve-radio"><span class="ve-radio-container ve-radio-checked"><input class="ve-radio-input" type="radio"><span class="ve-radio-inner"></span></span><span class="ve-radio-label">controlled</span></label></div>
</div>
`;
